org.lflang/src/org/lflang/generator/GeneratorUtils.java

```diff
@@ -55,16 +56,14 @@ private GeneratorUtils() {
      * Return the target declaration found in the given resource.
      */
     public static TargetDecl findTarget(Resource resource) {
+        int cnt = 0;
         TargetDecl targetDecl = null;
         for (TargetDecl t : findAll(resource, TargetDecl.class)) { // getAllContents should never return null.
-            if (targetDecl != null) {
-                throw new InvalidSourceException("There is more than one target!"); // FIXME: check this in validator
-            }
             targetDecl = t;
+            cnt++;
         }
-        if (targetDecl == null) {
-            throw new InvalidSourceException("No target found!");
-        }
+        if (cnt > 1)  throw new InvalidSourceException("There is more than one target!"); // FIXME: check this in validator
```

Comment:
```suggestion
```
I don't think our grammar permits more than one target declaration, so this check is unnecessary altogether.